logbook-ktor-server/src/main/kotlin/org/zalando/logbook/server/LogbookServer.kt

```diff
@@ -27,22 +28,24 @@ class LogbookServer(
         var logbook: Logbook = Logbook.create()
     }
 
-    companion object : ApplicationFeature<Application, Config, LogbookServer> {
-        private val responseProcessingStageKey: AttributeKey<Logbook.ResponseProcessingStage> = AttributeKey("Logbook.ResponseProcessingStage")
+    companion object : BaseApplicationPlugin<Application, Config, LogbookServer> {
+        private val responseProcessingStageKey: AttributeKey<Logbook.ResponseProcessingStage> =
+            AttributeKey("Logbook.ResponseProcessingStage")
         override val key: AttributeKey<LogbookServer> = AttributeKey("LogbookServer")
         override fun install(pipeline: Application, configure: Config.() -> Unit): LogbookServer {
             val config = Config().apply(configure)
-            val feature = LogbookServer(config.logbook)
+            val plugin = LogbookServer(config.logbook)
 
             pipeline.receivePipeline.intercept(ApplicationReceivePipeline.Before) {
                 val request = ServerRequest(call.request)
-                val requestWritingStage = feature.logbook.process(request)
+                val requestWritingStage = plugin.logbook.process(request)
                 val proceedWith = when {
                     request.shouldBuffer() && !call.request.receiveChannel().isClosedForRead -> {
                         val content = call.request.receiveChannel().readBytes()
                         request.buffer(content)
-                        ApplicationReceiveRequest(it.typeInfo, ByteReadChannel(content))
+                        ByteReadChannel(content)
```

Comment:
This line needs revision.
I just put something here so that the class compiles and the tests pass.

Reply:
plugins work different these days. I created a new branch from the current main. I also try to adapt the tests to the new API

Reply:
This also seems right according to the migration docs - https://ktor.io/docs/migrating-2.html#serialization-custom-converter

Reply:
Or I misunderstood it 🤔